<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_cdn_endpoint_invalid_querystring_caching_behaviour

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

Allowed values are:
- IgnoreQueryString
- BypassCaching
- UseQueryString
- NotSet

## Example

```hcl
resource "azurerm_cdn_endpoint" "foo" {
  querystring_caching_behaviour = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: "..." is an invalid value as querystring_caching_behaviour (azurerm_cdn_endpoint_invalid_querystring_caching_behaviour)

  on template.tf line 2:
  2:   querystring_caching_behaviour = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_cdn_endpoint_invalid_querystring_caching_behaviour.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/cdn/resource-manager/Microsoft.Cdn/stable/2020-09-01/cdn.json
