<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_data_factory_trigger_schedule_invalid_frequency

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

Allowed values are:
- NotSpecified
- Minute
- Hour
- Day
- Week
- Month
- Year

## Example

```hcl
resource "azurerm_data_factory_trigger_schedule" "foo" {
  frequency = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: "..." is an invalid value as frequency (azurerm_data_factory_trigger_schedule_invalid_frequency)

  on template.tf line 2:
  2:   frequency = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_data_factory_trigger_schedule_invalid_frequency.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/datafactory/resource-manager/Microsoft.DataFactory/DataFactory/stable/2018-06-01/entityTypes/Trigger.json
