<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_traffic_manager_profile_invalid_profile_status

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

Allowed values are:
- Enabled
- Disabled

## Example

```hcl
resource "azurerm_traffic_manager_profile" "foo" {
  profile_status = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: "..." is an invalid value as profile_status (azurerm_traffic_manager_profile_invalid_profile_status)

  on template.tf line 2:
  2:   profile_status = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_traffic_manager_profile_invalid_profile_status.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/trafficmanager/resource-manager/Microsoft.Network/TrafficManager/stable/2022-04-01/trafficmanager.json
