<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_image_invalid_hyper_v_generation

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

Allowed values are:
- V1
- V2

## Example

```hcl
resource "azurerm_image" "foo" {
  hyper_v_generation = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: "..." is an invalid value as hyper_v_generation (azurerm_image_invalid_hyper_v_generation)

  on template.tf line 2:
  2:   hyper_v_generation = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_image_invalid_hyper_v_generation.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/compute.json
