<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_kusto_eventhub_data_connection_invalid_data_format

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

Allowed values are:
- MULTIJSON
- JSON
- CSV
- TSV
- SCSV
- SOHSV
- PSV
- TXT
- RAW
- SINGLEJSON
- AVRO
- TSVE
- PARQUET
- ORC
- APACHEAVRO
- W3CLOGFILE

## Example

```hcl
resource "azurerm_kusto_eventhub_data_connection" "foo" {
  data_format = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: "..." is an invalid value as data_format (azurerm_kusto_eventhub_data_connection_invalid_data_format)

  on template.tf line 2:
  2:   data_format = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_kusto_eventhub_data_connection_invalid_data_format.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2021-08-27/kusto.json
