<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_maintenance_configuration_invalid_scope

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

Allowed values are:
- Host
- Resource
- OSImage
- Extension
- InGuestPatch
- SQLDB
- SQLManagedInstance

## Example

```hcl
resource "azurerm_maintenance_configuration" "foo" {
  scope = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: "..." is an invalid value as scope (azurerm_maintenance_configuration_invalid_scope)

  on template.tf line 2:
  2:   scope = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_maintenance_configuration_invalid_scope.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/maintenance/resource-manager/Microsoft.Maintenance/preview/2022-07-01-preview/Maintenance.json
