<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_virtual_network_gateway_invalid_sku

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

Allowed values are:
- Basic
- HighPerformance
- Standard
- UltraPerformance
- VpnGw1
- VpnGw2
- VpnGw3
- VpnGw4
- VpnGw5
- VpnGw1AZ
- VpnGw2AZ
- VpnGw3AZ
- VpnGw4AZ
- VpnGw5AZ
- ErGw1AZ
- ErGw2AZ
- ErGw3AZ
- ErGwScale

## Example

```hcl
resource "azurerm_virtual_network_gateway" "foo" {
  sku = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: "..." is an invalid value as sku (azurerm_virtual_network_gateway_invalid_sku)

  on template.tf line 2:
  2:   sku = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_virtual_network_gateway_invalid_sku.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/virtualNetworkGateway.json
