<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_kubernetes_cluster_node_pool_invalid_os_disk_size_gb

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

The rule requires the value to be in the range 0 <= x <= 1023.

## Example

```hcl
resource "azurerm_kubernetes_cluster_node_pool" "foo" {
  os_disk_size_gb = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: os_disk_size_gb must be 1023 or less (azurerm_kubernetes_cluster_node_pool_invalid_os_disk_size_gb)

  on template.tf line 2:
  2:   os_disk_size_gb = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_kubernetes_cluster_node_pool_invalid_os_disk_size_gb.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-04-01/managedClusters.json
