<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_cosmosdb_account_invalid_offer_type

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

Allowed values are:
- Standard

## Example

```hcl
resource "azurerm_cosmosdb_account" "foo" {
  offer_type = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: "..." is an invalid value as offer_type (azurerm_cosmosdb_account_invalid_offer_type)

  on template.tf line 2:
  2:   offer_type = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_cosmosdb_account_invalid_offer_type.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2023-04-15/cosmos-db.json
