<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_packet_capture_invalid_maximum_bytes_per_packet

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

The rule requires the value to be in the range 0 <= x <= 4294967295.

## Example

```hcl
resource "azurerm_packet_capture" "foo" {
  maximum_bytes_per_packet = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: maximum_bytes_per_packet must be 4294967295 or less (azurerm_packet_capture_invalid_maximum_bytes_per_packet)

  on template.tf line 2:
  2:   maximum_bytes_per_packet = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_packet_capture_invalid_maximum_bytes_per_packet.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/networkWatcher.json
