<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_search_service_invalid_replica_count

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

The rule requires the value to be in the range 1 <= x <= 12.

## Example

```hcl
resource "azurerm_search_service" "foo" {
  replica_count = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: replica_count must be 12 or less (azurerm_search_service_invalid_replica_count)

  on template.tf line 2:
  2:   replica_count = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_search_service_invalid_replica_count.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/search/resource-manager/Microsoft.Search/stable/2022-09-01/search.json
