<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_netapp_pool_invalid_service_level

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

Allowed values are:
- Standard
- Premium
- Ultra
- StandardZRS

## Example

```hcl
resource "azurerm_netapp_pool" "foo" {
  service_level = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: "..." is an invalid value as service_level (azurerm_netapp_pool_invalid_service_level)

  on template.tf line 2:
  2:   service_level = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_netapp_pool_invalid_service_level.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/netapp/resource-manager/Microsoft.NetApp/stable/2021-10-01/netapp.json
