<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_automation_runbook_invalid_runbook_type

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

Allowed values are:
- Script
- Graph
- PowerShellWorkflow
- PowerShell
- GraphPowerShellWorkflow
- GraphPowerShell
- Python2
- Python3
- PowerShell72

## Example

```hcl
resource "azurerm_automation_runbook" "foo" {
  runbook_type = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: "..." is an invalid value as runbook_type (azurerm_automation_runbook_invalid_runbook_type)

  on template.tf line 2:
  2:   runbook_type = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_automation_runbook_invalid_runbook_type.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/automation/resource-manager/Microsoft.Automation/stable/2023-11-01/runbook.json
