<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_log_analytics_workspace_invalid_retention_in_days

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

The rule requires the value to be in the range -1 <= x <= 730.

## Example

```hcl
resource "azurerm_log_analytics_workspace" "foo" {
  retention_in_days = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: retention_in_days must be 730 or less (azurerm_log_analytics_workspace_invalid_retention_in_days)

  on template.tf line 2:
  2:   retention_in_days = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_log_analytics_workspace_invalid_retention_in_days.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2020-03-01-preview/Workspaces.json
