<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_traffic_manager_endpoint_invalid_endpoint_status

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

Allowed values are:
- Enabled
- Disabled

## Example

```hcl
resource "azurerm_traffic_manager_endpoint" "foo" {
  endpoint_status = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: "..." is an invalid value as endpoint_status (azurerm_traffic_manager_endpoint_invalid_endpoint_status)

  on template.tf line 2:
  2:   endpoint_status = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_traffic_manager_endpoint_invalid_endpoint_status.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/trafficmanager/resource-manager/Microsoft.Network/stable/2018-04-01/trafficmanager.json
