<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_kusto_database_principal_invalid_role

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

Allowed values are:
- Admin
- Ingestor
- Monitor
- User
- UnrestrictedViewer
- Viewer

## Example

```hcl
resource "azurerm_kusto_database_principal" "foo" {
  role = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: "..." is an invalid value as role (azurerm_kusto_database_principal_invalid_role)

  on template.tf line 2:
  2:   role = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_kusto_database_principal_invalid_role.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2021-01-01/kusto.json
