<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_iothub_endpoint_storage_container_invalid_batch_frequency_in_seconds

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

The rule requires the value to be in the range 60 <= x <= 720.

## Example

```hcl
resource "azurerm_iothub_endpoint_storage_container" "foo" {
  batch_frequency_in_seconds = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: batch_frequency_in_seconds must be 720 or less (azurerm_iothub_endpoint_storage_container_invalid_batch_frequency_in_seconds)

  on template.tf line 2:
  2:   batch_frequency_in_seconds = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_iothub_endpoint_storage_container_invalid_batch_frequency_in_seconds.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/iothub/resource-manager/Microsoft.Devices/stable/2021-03-31/iothub.json
