<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_eventhub_namespace_invalid_capacity

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

The rule requires the value to be 0 or higher.

## Example

```hcl
resource "azurerm_eventhub_namespace" "foo" {
  capacity = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: capacity must be 0 or higher (azurerm_eventhub_namespace_invalid_capacity)

  on template.tf line 2:
  2:   capacity = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_eventhub_namespace_invalid_capacity.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/namespaces-preview.json
