<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_dedicated_host_invalid_platform_fault_domain

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

The rule requires the value to be 0 or higher.

## Example

```hcl
resource "azurerm_dedicated_host" "foo" {
  platform_fault_domain = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: platform_fault_domain must be 0 or higher (azurerm_dedicated_host_invalid_platform_fault_domain)

  on template.tf line 2:
  2:   platform_fault_domain = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_dedicated_host_invalid_platform_fault_domain.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/compute/resource-manager/Microsoft.Compute/stable/2021-11-01/compute.json
