<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_key_vault_key_invalid_curve

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

Allowed values are:
- P-256
- P-384
- P-521
- P-256K

## Example

```hcl
resource "azurerm_key_vault_key" "foo" {
  curve = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: "..." is an invalid value as curve (azurerm_key_vault_key_invalid_curve)

  on template.tf line 2:
  2:   curve = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_key_vault_key_invalid_curve.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.4/keys.json
