<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_hdinsight_kafka_cluster_invalid_tier

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

Allowed values are:
- Standard
- Premium

## Example

```hcl
resource "azurerm_hdinsight_kafka_cluster" "foo" {
  tier = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: "..." is an invalid value as tier (azurerm_hdinsight_kafka_cluster_invalid_tier)

  on template.tf line 2:
  2:   tier = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_hdinsight_kafka_cluster_invalid_tier.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/hdinsight/resource-manager/Microsoft.HDInsight/HDInsight/stable/2021-06-01/cluster.json
