<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_healthcare_service_invalid_kind

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

Allowed values are:
- fhir
- fhir-Stu3
- fhir-R4

## Example

```hcl
resource "azurerm_healthcare_service" "foo" {
  kind = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: "..." is an invalid value as kind (azurerm_healthcare_service_invalid_kind)

  on template.tf line 2:
  2:   kind = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_healthcare_service_invalid_kind.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/healthcareapis/resource-manager/Microsoft.HealthcareApis/HealthcareApis/stable/2022-12-01/healthcare-apis.json
