<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_healthcare_service_invalid_cosmosdb_throughput

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

The rule requires the value to be 400 or higher.

## Example

```hcl
resource "azurerm_healthcare_service" "foo" {
  cosmosdb_throughput = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: cosmosdb_throughput must be 400 or higher (azurerm_healthcare_service_invalid_cosmosdb_throughput)

  on template.tf line 2:
  2:   cosmosdb_throughput = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_healthcare_service_invalid_cosmosdb_throughput.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/healthcareapis/resource-manager/Microsoft.HealthcareApis/stable/2022-12-01/healthcare-apis.json
