<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_iothub_endpoint_storage_container_invalid_encoding

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

Allowed values are:
- Avro
- AvroDeflate
- JSON

## Example

```hcl
resource "azurerm_iothub_endpoint_storage_container" "foo" {
  encoding = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: "..." is an invalid value as encoding (azurerm_iothub_endpoint_storage_container_invalid_encoding)

  on template.tf line 2:
  2:   encoding = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_iothub_endpoint_storage_container_invalid_encoding.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/iothub/resource-manager/Microsoft.Devices/stable/2021-03-31/iothub.json
