<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_storage_account_network_rules_invalid_default_action

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

Allowed values are:
- Allow
- Deny

## Example

```hcl
resource "azurerm_storage_account_network_rules" "foo" {
  default_action = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: "..." is an invalid value as default_action (azurerm_storage_account_network_rules_invalid_default_action)

  on template.tf line 2:
  2:   default_action = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_storage_account_network_rules_invalid_default_action.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/storage/resource-manager/Microsoft.Storage/stable/2021-04-01/storage.json
