<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_container_registry_webhook_invalid_registry_name

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

In this rule, the string must match the regular expression `^[a-zA-Z0-9]*$``.

## Example

```hcl
resource "azurerm_container_registry_webhook" "foo" {
  registry_name = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: "..." does not match valid pattern ^[a-zA-Z0-9]*$ (azurerm_container_registry_webhook_invalid_registry_name)

  on template.tf line 2:
  2:   registry_name = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_container_registry_webhook_invalid_registry_name.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2021-08-01-preview/containerregistry.json
