<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_managed_disk_invalid_storage_account_type

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

Allowed values are:
- Standard_LRS
- Premium_LRS
- StandardSSD_LRS
- UltraSSD_LRS
- Premium_ZRS
- StandardSSD_ZRS
- PremiumV2_LRS

## Example

```hcl
resource "azurerm_managed_disk" "foo" {
  storage_account_type = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: "..." is an invalid value as storage_account_type (azurerm_managed_disk_invalid_storage_account_type)

  on template.tf line 2:
  2:   storage_account_type = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_managed_disk_invalid_storage_account_type.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/compute/resource-manager/Microsoft.Compute/DiskRP/stable/2022-03-02/disk.json
