<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_redis_cache_invalid_private_static_ip_address

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

In this rule, the string must match the regular expression `^\d+\.\d+\.\d+\.\d+$``.

## Example

```hcl
resource "azurerm_redis_cache" "foo" {
  private_static_ip_address = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: "..." does not match valid pattern ^\d+\.\d+\.\d+\.\d+$ (azurerm_redis_cache_invalid_private_static_ip_address)

  on template.tf line 2:
  2:   private_static_ip_address = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_redis_cache_invalid_private_static_ip_address.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/redis/resource-manager/Microsoft.Cache/stable/2023-08-01/redis.json
