<!--- This file generated by `tools/apispec-rule-gen/main.go`. DO NOT EDIT --->

# azurerm_database_migration_project_invalid_source_platform

Warns about values that appear to be invalid based on [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs).

Allowed values are:
- SQL
- Unknown

## Example

```hcl
resource "azurerm_database_migration_project" "foo" {
  source_platform = ... // invalid value
}
```

```
$ tflint
1 issue(s) found:

Error: "..." is an invalid value as source_platform (azurerm_database_migration_project_invalid_source_platform)

  on template.tf line 2:
  2:   source_platform = ... // invalid value

Reference: https://github.com/terraform-linters/tflint-ruleset-azurerm/blob/v0.1.0/docs/rules/azurerm_database_migration_project_invalid_source_platform.md

```

## Why

Requests containing invalid values will return an error when calling the API by `terraform apply`.

## How to Fix

Replace the warned value with a valid value.

## Source

This rule is automatically generated from [azure-rest-api-specs](https://github.com/Azure/azure-rest-api-specs). If you are uncertain about the warning, check the following API schema referenced by this rule.

https://github.com/Azure/azure-rest-api-specs/tree/master/specification/datamigration/resource-manager/Microsoft.DataMigration/stable/2018-04-19/definitions/Projects.json
